### PR TITLE
Extraction pro

### PR DESCRIPTION
**Pull request by Michaelyya**
-- ADDED generate_prompt() & gemini_analyze_topics() & fetch_html()

Test it in "Gemini-Scraper/gemini/analyze_content.py"

return 

" 
"Early days of the company": "Steve Jobs and Steve Wozniak founded Apple Computer in 1976 to sell the Apple I personal computer. The company was incorporated in 1977 and went public in 1980. In the early 1980s, Apple introduced the Apple II series of personal computers, which was a commercial success. However, the company faced increasing competition from IBM and other PC manufacturers in the mid-1980s.",
  "List of products": [
    "Apple I",
    "Apple II",
    "Apple III",
    "Apple Lisa",
    "Macintosh",
    "iMac",
    "iPod",
    "iPhone",
    "iPad",
    "Apple Watch",
    "Apple TV",
    "Apple HomePod"
  ],
  "Important people in the company": [
    "Steve Jobs",
    "Steve Wozniak",
    "Tim Cook",
    "Jony Ive",
    "Phil Schiller",
    "Craig Federighi",
    "Eddy Cue",
    "Jeff Williams",
    "Johny Srouji"
  ]
}
"

### CODE REVIEW

File: gemini/analyze_content.py [hunk 1]
```diff
@@ -26,23 +29,59 @@ def tag_visible(element):
...
     return [
         html_content[i : i + chunk_size]
         for i in range(0, len(html_content), chunk_size)
     ]
 
+def generate_prompt(topics, text_chunk):
+    example_json = {
+        topics[0]: "Relevant information about topic one.",
+        topics[1]: "Relevant information about topic two.",
+        topics[2]: "Relevant information about topic three."
```
Comment (Marvin-Deng) [inline gemini/analyze_content.py:44]: Try to generalize this logic for any number of topics. If there are multiple topics that person wants to search for, then you would only be getting 3 topics. If there are less than 3 topics, there would be an index error

File: gemini/analyze_content.py [hunk 2]
```diff
@@ -26,23 +29,59 @@ def tag_visible(element):
...
+    }
+    return f"""
+    Please analyze the following text and categorize the information according to these topics: {', '.join(topics)}.
+    For each topic, format the information in JSON as shown in this example: {json.dumps(example_json, indent=2)}.
+
```
Comment (Marvin-Deng) [inline gemini/analyze_content.py:49]: Reformat the return string with indentations

File: main.py [hunk 3]
```diff
@@ -5,6 +5,13 @@
 
 
 def main():
+    # json_format = """
+    # {
+    # "topics": str,
+    # "info": str,
+    # }
+    # """
+
```
Comment (Marvin-Deng) [inline main.py:14]: Remove commented json

**Review by Marvin-Deng — CHANGES_REQUESTED** [review]

**Comment by Michaelyya** [comment]
Checked and updated

**Review by Marvin-Deng — APPROVED** [review]

**Pull request merged**